### PR TITLE
Attempt to resolve indexer conflict

### PR DESCRIPTION
**Pull request by Oats87**
I have no idea if this is going to work for https://github.com/rancher/rancher/issues/34971

Part of the work for https://github.com/rancher/rancher/issues/36162

### CODE REVIEW

**Review by rosskirkpat — DISMISSED** [review]
LGTM

File: pkg/rancher/migrations.go [hunk 1]
```diff
@@ -239,26 +241,32 @@ func addWebhookConfigToCAPICRDs(crdClient controllerapiextv1.CustomResourceDefin
...
 	for _, crd := range crds.Items {
-		if crd.Spec.Group != "cluster.x-k8s.io" || (crd.Spec.Conversion != nil && crd.Spec.Conversion.Strategy != apiextv1.NoneConverter) {
-			continue
-		}
+		if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			latestCrd, err := crdClient.Get(crd.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if latestCrd.Spec.Group != "cluster.x-k8s.io" || (latestCrd.Spec.Conversion != nil && latestCrd.Spec.Conversion.Strategy != apiextv1.NoneConverter) {
```
Comment (thedadams) [inline pkg/rancher/migrations.go:249]: Let's move this outside the retry because this won't change and we won't `Get` if we don't need to.

Comment (Oats87) [reply pkg/rancher/migrations.go]: Moved

**Review by rosskirkpat — APPROVED** [review]

**Review by thedadams — APPROVED** [review]

**Comment by Oats87** [comment]
https://github.com/rancher/rancher/issues/36162

**Pull request merged**